x-pack/plugins/apm/server/plugin.ts

```diff
@@ -29,7 +29,7 @@ export class APMPlugin implements Plugin<APMPluginContract> {
     this.legacySetup$ = new AsyncSubject();
   }
 
-  public setup(
+  public async setup(
```

Comment:
Will kibana actually wait for async plugins?

Reply:
Right now I think it has to, because other plugins could depend on the setup contract returned by the `setup` function. That might change (per https://github.com/elastic/kibana/pull/45796), but I'm not sure what the API would then look like.